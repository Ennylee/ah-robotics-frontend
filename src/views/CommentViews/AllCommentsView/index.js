import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';
import CommentsListComponent from '../../../components/Comments/articleComments';
import { getArticleComments } from '../../../redux/actions/CommentActions/actions';

class AllCommentsView extends Component {
  constructor(props) {
    super(props);
    this.state = { };
  }

  onChange = (event) => {
    this.setState({ [event.target.name]: event.target.value });
  };

  render() {
    const { getComments } = this.props;
    const commentsList = getComments.comments.comment;
    const totalCount = getComments.comments.commentsCount;
    const loading = getComments.isLoading;
    return (
      <div>
        <CommentsListComponent comments={commentsList} count={totalCount} loading={loading} />
      </div>
    );
  }
}

AllCommentsView.propTypes = {
  getComments: PropTypes.shape({}).isRequired,
};

const mapStateToProps = (
  { getComments, addArticleComment },
) => (
    { getComments, addArticleComment });

const mapDispatchToProps = dispatch => bindActionCreators({ getArticleComments }, dispatch);


export default connect(mapStateToProps, mapDispatchToProps)(AllCommentsView);
