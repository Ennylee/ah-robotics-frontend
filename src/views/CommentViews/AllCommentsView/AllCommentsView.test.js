import React from 'react';
import { shallow } from 'enzyme';
import AllCommentsView from './index';

describe('All Comments View', () => {
  it('view should match the snapshot', () => {
    const wrapper = shallow(<AllCommentsView />);
    expect(wrapper).toMatchSnapshot();
  });

  it('app should render AllCommentsView page', () => {
    const wrapper = shallow(<AllCommentsView />);
    expect(wrapper.length).toBe(1);
  });
});
