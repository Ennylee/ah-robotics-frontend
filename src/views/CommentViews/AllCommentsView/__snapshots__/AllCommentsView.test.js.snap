// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All Comments View view should match the snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
